### PR TITLE
Apply BaseModelWrapper inside of BaseModelCallback

### PR DESCRIPTION
**Pull request by DanteOz**
Wrap HuggingFace PreTrainedModel during before_fit() and after_fit() callback hooks. This eliminates the need to explicitly wrap the PreTrainedModel in the low level api.

Tasks:
- [x] Wrap/unwrap model in BaseModelCallback
- [x] Maintain backwards compatibility by checking if model is already wrapped.
- [x] Ensure BaseModelCallback is first callback added when initializing Learner.

### CODE REVIEW

File: nbs/11_text-modeling-core.ipynb [hunk 1]
```diff
@@ -80,9 +80,9 @@
      "output_type": "stream",
```
[the remark this answers is not in the document]
Comment (DanteOz) [reply nbs/11_text-modeling-core.ipynb:80]: Updated in recent commit.

**Comment by ohmeow** [comment]
Looking at this today and tomorrow ... sorry for the delay :)

**Comment by ohmeow** [comment]
Thanks much sir!  Yah ... its being merged ... the miracle is happening :)

**Comment by ohmeow** [comment]
Merged

**Pull request merged**